src/crypto.js

```diff
@@ -76,7 +76,7 @@ export default function makeCryptoWith(opts) {
   _validateOpts(opts);
   const { encryptionKey } = opts;
 
-  function encrypt(input) {
+  function encrypt(input, aad) {
```

Comment:
Can we use the same serialization approach which we use for the `input` with the `aad`?

Reply:
The problem is guaranteeing ordering of keys and array elements with `JSON.stringify` we'd have to use something like `object-hash` or create one for use here. The aad buffer needs to be exactly the same bytes. That can be done by the user of this package prior to passing in the AAD.

Reply:
If we're going to require it to be a string, can we add tests for this and ensure a user comprehensible Error is thrown?

Reply:
@kobelb Ok I added a validation step for the AAD if it is defined.